### PR TITLE
Cleanup old schema builder; expose minimal set of new helper methods.

### PR DESCRIPTION
**Pull request by Shane32**
Depends on
- #2977 

1. Remove:
- `EventStreamResolver`
- `AsyncEventStreamResolver`
- `DelegateFieldModelBinderResolver`
- `ReflectionHelper.BuildArguments`

2. Expose:
- `MemberResolver`
- `EventStreamMethodResolver`
- `AutoRegisteringHelper.BuildFieldResolver`
- `AutoRegisteringHelper.BuildEventStreamResolver`

3. Add migration notes.

### CODE REVIEW

**Comment by sungam3r** [comment]
изображение
Member/Method . Right naming?

**Comment by Shane32** [comment]
Yes the member resolver works for fields, properties or methods. The event stream methods resolver works only for methods.

**Comment by Shane32** [comment]
I was thinking of merging those two classes. I’ll probably write a PR and we can discuss if that’s a good or bad idea. Or you can let me know your thoughts now.  I’m okay either way. So far I’m assuming they are better as separate classes.

**Comment by Shane32** [comment]
@sungam3r This PR is ready for review.

**Comment by Shane32** [comment]
@sungam3r We seem to be getting a lot of intermittent build check failures. I’ve had to run the tests again for the last few PRs.

**Comment by sungam3r2022** [comment]
Good news - despite of blocking I still get email notifications.

File: docs2/site/docs/migrations/migration5.md [hunk 1]
```diff
@@ -784,3 +784,21 @@ See New Features: 'Schema builder and `FieldDelegate` improvements for reflected
...
 See New Features: 'Schema builder and `FieldDelegate` improvements for reflected methods' above.
+
+### 32. Code removed to support prior implementation of FieldDelegate and schema builder
+
+The following classes and methods have been removed:
+
+- The `EventStreamResolver` implementation which accepted an `IAccessor` as a construtor parameter has been removed.
+- The `AsyncEventStreamResolver` implementation which accepted an `IAccessor` as a construtor parameter has been removed.
+- The `DelegateFieldModelBinderResolver` class has been removed.
+- The `ReflectionHelper.BuildArguments` method has been removed.
```
Comment (sungam3r2022) [inline docs2/site/docs/migrations/migration5.md:795]:
```suggestion
- The `EventStreamResolver` implementation which accepted an `IAccessor` as a construtor parameter.
- The `AsyncEventStreamResolver` implementation which accepted an `IAccessor` as a construtor parameter.
- The `DelegateFieldModelBinderResolver` class.
- The `ReflectionHelper.BuildArguments` method.
```

File: docs2/site/docs/migrations/migration5.md [hunk 2]
```diff
@@ -784,3 +784,21 @@ See New Features: 'Schema builder and `FieldDelegate` improvements for reflected
...
+
+You may use the following classes and methods as replacements:
+
+- The `MemberResolver` class is an `IFieldResolver` implementation for a property, method or field. Expressions are passed
+  to the constructor for the instance (and if applicable, method arguments), which is immediately compiled.
+- The `EventStreamMethodResolver` class is an `IEventStreamResolver` implementation for a method that returns an `IObservable<T>`
+  or `Task<IObservable<T>>`. It also implements provides a basic `IFieldResolver` implementation for subscription fields.
```
Comment (sungam3r2022) [inline docs2/site/docs/migrations/migration5.md:802]: > implements provides

?

**Review by sungam3r2022 — APPROVED** [review]

File: docs2/site/docs/migrations/migration5.md (hunk 2 above)
Comment (Shane32) [reply docs2/site/docs/migrations/migration5.md:802]:
```suggestion
  or `Task<IObservable<T>>`. It also provides a basic `IFieldResolver` implementation for subscription fields.
```

**Pull request merged**